### PR TITLE
test

### PR DESCRIPTION
**Pull request by zhangrui1203**

### CODE REVIEW

File: test [hunk 1]
```diff
@@ -0,0 +1,478 @@
...
+static struct termios saved_tty;
+static long delay = 3;
+static int (*sort_func)(const struct irq_info *, const struct irq_info *);
+static long smp_num_cpus;
+static char *program;
+
+/*
+ * irqinfo - parse the system's interrupts
+ */
+static struct irq_stat *get_irqinfo()
```
Comment (pizhenwei) [inline test:80]: LGTM.

File: test [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+	struct irq_stat *stat;
+	struct irq_info *curr;
+
+	/* NAME + ':' + 11 bytes/cpu + IRQ_DESC_LEN */
+	bufferlen = IRQ_NAME_LEN + 1 + smp_num_cpus * 11 + IRQ_DESC_LEN;
+	buffer = malloc(bufferlen);
+	if (!buffer)
+		goto out;
+	
+	stat = calloc(1, sizeof(*stat));
```
Comment (pizhenwei) [inline test:94]: LGTM2.

File: test [hunk 3]
```diff
@@ -0,0 +1,478 @@
...
+			tmp += 11;
+		}
+
+		if (tmp - buffer < length) {
+			/* strip all space before desc */
+			while(*tmp == ' ')
+				tmp++;
+			strcpy(curr->desc, tmp);
+		} else {
+			/* no desc string at all, we have to set '\n' here */
```
Comment (pizhenwei) [inline test:154]: LGTM3

File: test (hunk 1 above)
Comment (zhangrui1203) [reply test:80]: gadfadfafd

File: test (hunk 2 above)
Comment (zhangrui1203) [reply test:94]: asdfa dsfafasdfas

File: test [hunk 4]
```diff
@@ -0,0 +1,478 @@
...
+#include <ncurses.h>
+#include <termios.h>
+#include <getopt.h>
+#include <ctype.h>
+#include <sys/ioctl.h>
+
+#include <sys/select.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <unistd.h>
```
Comment (pizhenwei) [inline test:39]: test!

**Review by pizhenwei — CHANGES_REQUESTED** [review]

**Pull request closed without merging**